pycbc/events/significance.py

```diff
@@ -0,0 +1,253 @@
+    inputs
+    """
+    calculation_method = method_dict['method']
+    fit_func = method_dict['function']
+    fit_thresh = method_dict['threshold']
+    return _significance_meth_dict[calculation_method](back_stat, fore_stat,
+                                                       dec_facs,
+                                                       fit_func=fit_func,
+                                                       fit_thresh=fit_thresh)
+
+
+def insert_significance_option_group(parser):
+    """
+    Add some options for use when a significance is being estimated from
+    events or event distributions.
+    """
+    parser.add_argument('--far-calculation-method', nargs='+',
+                        default=[],
+                        help="Method used for FAR calculation in each "
+                             "detector combination, given as "
+                             "combination:method pairs, i.e. "
+                             "H1:trigger_fit H1L1:n_louder H1L1V1:n_louder "
+                             "etc. Method options are ["
+                             + ",".join(_significance_meth_dict.keys()) +
+                             "]. Default = n_louder for all not given")
+    parser.add_argument('--fit-threshold', nargs='+', default=[],
+                        help="Thresholds for the fits to statistic "
+                             "values for FAN approximation if "
+                             "--far-calculation-method is 'trigger_fit'. "
+                             "Given as combination-values pairs, e.g. "
+                             "H1:0 L1:0 V1:-4, for all combinations which "
+                             "have --far-calculation-method "
+                             "of trigger_fit")
+    parser.add_argument("--fit-function", nargs='+', default=[],
+                        help="Functional form for the statistic slope fit if "
+                             "--far-calculation-method is 'trigger_fit'. "
+                             "Given as combination:function pairs, i.e. "
+                             "H1:exponential H1L1:n_louder H1L1V1:n_louder. "
+                             "Options: ["
+                             + ",".join(trstats.fitalpha_dict.keys()) + "]. "
+                             "Default = exponential for all")
+
+
+def check_significance_options(args, parser):
+    """
+    Check that the arguments given for the insert_significance_option_group
+    options make sense
+    """
+    # Check that the key:method/function/threshold are in the
+    # right format, and are in allowed combinations
+    lists_to_check = [(args.far_calculation_method, str,
+                       _significance_meth_dict.keys()),
+                      (args.fit_function, str,
+                       trstats.fitalpha_dict.keys()),
+                      (args.fit_threshold, float,
+                       range(-10000, 10000))]
+    # (10,000 is suitable for current use but can be changed if wanted)
```

Comment:
the comment isn't so clear IMO - you are putting in hard-coded values which can only be changed by making new commits. Plus what we mean by 'current' use may not be constant over time - at some point the statement might become untrue. If the only reason to put in values here is to have _some_ check then I think we prefer -inf to inf.

Reply:
Tried that and -inf to inf comparison slows things down so to kill the program unfortunately.
I wanted to find a thing where we could use `if value in `, and the range seemed sensible

Reply:
I put in a special None case where the test isn't done